Build failure: python3.11-jedi-language-server (nixpkgs/master)

I have the build failure issue after one/some of today's commits with nixpkgs/master

### Build log

```
error: builder for '/nix/store/7l57swg6fkpk0124kc31qx3ppm3n27lr-python3.11-jedi-language-server-0.41.1-unstable-2023-10-04.drv' failed with exit code 1;
       last 10 log lines:
       > FAILED tests/lsp_tests/test_completion.py::test_lsp_completion_class_method - AssertionError:
       > FAILED tests/lsp_tests/test_completion.py::test_lsp_completion_class_noargs - AssertionError:
       > FAILED tests/lsp_tests/test_hover.py::test_hover_on_module - AssertionError:
       > FAILED tests/lsp_tests/test_hover.py::test_hover_on_function - AssertionError:
       > FAILED tests/lsp_tests/test_hover.py::test_hover_on_class - AssertionError:
       > FAILED tests/lsp_tests/test_hover.py::test_hover_on_method - AssertionError:
       > FAILED tests/lsp_tests/test_signature.py::test_signature_help[(-14-0] - AssertionError:
       > FAILED tests/lsp_tests/test_signature.py::test_signature_help[,-18-1] - AssertionError:
       > ======================== 10 failed, 42 passed in 22.92s ========================
       > /nix/store/c8dj731bkcdzhgrpawhc8qvdgls4xfjv-stdenv-linux/setup: line 1578: pop_var_context: head of shell_variables not a function context
       For full logs, run 'nix log /nix/store/7l57swg6fkpk0124kc31qx3ppm3n27lr-python3.11-jedi-language-server-0.41.1-unstable-2023-10-04.drv'.
```

### Additional context

One of failed tests:
```
            expected = {
                "label": "my_function",
                "kind": 3,
                "detail": "def my_function()",
                "documentation": {
                    "kind": "markdown",
                    "value": "```text\nSimple test function.\n```",
                },
                "sortText": "v0",
                "filterText": "my_function",
                "insertText": "my_function()$0",
                "insertTextFormat": 2,
            }
>           assert_that(actual, is_(expected))
E           AssertionError: 
E           Expected: <{'label': 'my_function', 'kind': 3, 'detail': 'def my_function()', 'documentation': {'kind': 'markdown', 'value': '```text\nSimple test function.\n```'}, 'sortText': 'v0', 'filterText': '>
E                but: was <{'label': 'my_function', 'kind': 3, 'detail': 'def my_function()', 'documentation': {'kind': 'markdown', 'value': 'Simple test function.'}, 'sortText': 'v0', 'filterText': 'my_functio>

tests/lsp_tests/test_completion.py:72: AssertionError
__________________________ test_eager_lsp_completion ___________________________
```
while original test does not contain this 'text\n...\n' multiline string
https://github.com/pappasam/jedi-language-server/blob/d64317fca56431c8b82b1ae2a54b09956c380a9b/tests/lsp_tests/test_completion.py#L62
```
                "value": "Simple test function.",
```

### Notify maintainers

@doronbehar @kalekseev 

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
spiage@nixos:~$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.8.1, NixOS, 24.05 (Uakari), 24.05.20240321.13a8da9`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.2`
 - channels(root): `""`
 - nixpkgs: `/nix/store/z9vp5l474dw9skn0scs9aqpbj6iyym0k-source`
```

---

--- a/pkgs/development/python-modules/jedi-language-server/default.nix
+++ b/pkgs/development/python-modules/jedi-language-server/default.nix
@@ -11,12 +11,11 @@
 , pytestCheckHook
 , python-lsp-jsonrpc
 , pythonOlder
-, pythonRelaxDepsHook
 }:
 
 buildPythonPackage rec {
   pname = "jedi-language-server";
-  version = "0.41.1-unstable-2023-10-04";
+  version = "0.41.3";
   format = "pyproject";
 
   disabled = pythonOlder "3.8";
@@ -24,17 +23,12 @@ buildPythonPackage rec {
   src = fetchFromGitHub {
     owner = "pappasam";
     repo = pname;
-    rev = "c4c470cff67e54593a626b22d1b6b05e56fde3a3";
-    hash = "sha256-qFBni97B/GkabbznnZtWTG4dCHFkOx5UQjuevxq+Uvo=";
+    rev = "v${version}";
+    hash = "sha256-+k4WOoEbVe7mlPyPj0ttBM+kmjq8V739yHi36BDYK2U=";
   };
-
-  pythonRelaxDeps = [
-    "pygls"
-  ];
 
   nativeBuildInputs = [
     poetry-core
-    pythonRelaxDepsHook
   ];
 
   propagatedBuildInputs = [

